app/src/main/java/org/simple/clinic/overdue/AppointmentCancelReason.kt

```diff
@@ -0,0 +1,74 @@
+package org.simple.clinic.overdue
+
+import android.arch.persistence.room.TypeConverter
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.ToJson
+
+sealed class AppointmentCancelReason {
+
+  object PatientNotResponding : AppointmentCancelReason()
+
+  object Moved : AppointmentCancelReason()
+
+  object Dead : AppointmentCancelReason()
+
+  object Other : AppointmentCancelReason()
+
+  data class Unknown(val actualValue: String) : AppointmentCancelReason()
+
+  /**
+   * It'll be nice to write a code generator instead of hand-typing
```

Comment:
We have many enums in the api, and will have many in the future. We might want to use this for some, or all of them. Makes sense to invest some time, either in this or a generic wrapper that will manage this, at some point. Let's revisit this when we have to convert our next enum to a string.

Reply:
Agreed.